### PR TITLE
DRAFT: stored procedures to migrate event to v2

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/analytics/private/sqls/redshift/fn-parse-utm-from-url.sql [hunk 1]
```diff
@@ -0,0 +1,283 @@
+
+create or replace function testapp002.parse_utm_from_url(url varchar(65535), referrer varchar(65535))
```
Comment (llmin) [inline src/analytics/private/sqls/redshift/fn-parse-utm-from-url.sql:2]: hard code

File: src/analytics/private/sqls/redshift/fn-rm-object-props.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+create or replace function testapp002.rm_object_props(o varchar(65535), props_list varchar(65535))
```
Comment (llmin) [inline src/analytics/private/sqls/redshift/fn-rm-object-props.sql:1]: hard code

File: src/analytics/private/sqls/redshift/sp-migrate-session-to-v2.sql [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+log_name VARCHAR(50) := 'sp_migrate_session_to_v2';
+
+BEGIN 
+
+DROP TABLE IF EXISTS tmp_traffic_session_web;
+
+DROP TABLE IF EXISTS tmp_traffic_session_mobile;
+
+INSERT INTO
+    testapp002.session(
```
Comment (llmin) [inline src/analytics/private/sqls/redshift/sp-migrate-session-to-v2.sql:19]: hard code

File: src/analytics/private/sqls/redshift/sp-migrate-event-to-v2.sql [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+    traffic_source.channel_group::varchar AS traffic_source_channel_group,
+    traffic_source.category::varchar AS traffic_source_category,
+    NULL AS user_first_touch_time_msec,
+    app_info.id::varchar AS app_package_id,
+    app_info.app_id::varchar AS app_id,
+    app_info.app_version::varchar AS app_version,
+    app_info.install_source::varchar AS app_install_source,
+CASE
+        WHEN platform is NULL THEN 'Web'
+        ELSE NULL
```
Comment (llmin) [inline src/analytics/private/sqls/redshift/sp-migrate-event-to-v2.sql:430]: ELSE NULL ?

File: src/analytics/private/sqls/redshift/sp-migrate-session-to-v2.sql [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+SELECT
+    user_pseudo_id,
+    session_id,
+    MAX(user_id),
+    MAX(session_number),
+    MIN(session_start_time_msec::bigint),
+    MAX(event_timestamp),
+    object('backfill_session', true)
+FROM
+    {{schema}}.event_v2 e
```
Comment (llmin) [inline src/analytics/private/sqls/redshift/sp-migrate-session-to-v2.sql:37]: It is better to make sure session_id is not null

and session_id is not null

Comment (amliuyong) [reply src/analytics/private/sqls/redshift/sp-migrate-session-to-v2.sql:49]: added

**Comment by amliuyong** [comment]
closed this PR

replace with https://github.com/awslabs/clickstream-analytics-on-aws/pull/1089

**Pull request closed without merging**